quipucords/api/inspectresult/model.py

```diff
@@ -3,16 +3,134 @@
 These models are used in the REST definitions
 """
 
-from django.db import models
+from django.db import connection, models
+from django.db.models.expressions import RawSQL
 from django.utils.translation import gettext as _
 
 from api import messages
 from api.common.util import RawFactEncoder
 from api.source.model import Source
+from constants import DataSources
+
+RAW_QUERY_POSTGRES = """
+with raw_agg as (
+    select
+        ir.id,
+        ir.result_set_id,
+        (
+        select
+            json_object_agg(raw.name, raw.value) as raw_facts
+        from
+            api_rawfact raw
+        where
+            raw.inspect_result_id = ir.id
```

Comment:
I wonder if we could pull this subquery into its parent query with a regular join. I may pull your branch and try some ideas.

What have you been doing to generate data for your benchmarks? I assume you're using the test factories. Would you share some specific commands that you used so I can follow the same steps for consistency?

Reply:
I did try that first, but it was slightly slower (both with real queries and with `EXPLAIN ANALYSE`).

Feel free to play around with it! I shared the data with you on slack, but I realized now that with the code changes introduced to factories we don't have control of how many "systems" we want to generate. So, in order to populate the db with synthetic data we would need to use the factories available on main branch and then migrate them.

Will see if I can quickly patch `ReportFactory` to enable this for the code in this branch.